config/new.go

```diff
@@ -74,6 +75,7 @@ func NewHCPConfig(opts ...HCPConfigOption) (HCPConfig, error) {
 			Scopes:      []string{"openid", "offline_access"},
 		},
 		session: &auth.UserSession{},
+		profile: profile.UserProfile{},
```

Comment:
Let's make this a pointer, so we're always mutating the same object if the profile needs to change.
```suggestion
		profile: &profile.UserProfile{},
```